Sources/Purchasing/Purchases/PurchasesOrchestrator.swift

```diff
@@ -933,16 +959,28 @@ private extension PurchasesOrchestrator {
                 return false
             }
 
-            callbacks[productIdentifier] = completion
+            callbacks[productIdentifier] = .init(completion)
             return true
         }
     }
 
     func getAndRemovePurchaseCompletedCallback(
         forTransaction transaction: StoreTransaction
     ) -> PurchaseCompletedBlock? {
-        return self.purchaseCompleteCallbacksByProductID.modify {
-            $0.removeValue(forKey: transaction.productIdentifier)
+        return self.purchaseCompleteCallbacksByProductID.modify { callbacks -> PurchaseCompletedBlock? in
+            guard let value = callbacks[transaction.productIdentifier] else { return nil }
+
+            if !transaction.hasKnownPurchaseDate || value.creationDate <= transaction.purchaseDate {
+                callbacks.removeValue(forKey: transaction.productIdentifier)
+                return value.completion
+            } else {
+                Logger.verbose(Strings.purchase.paymentqueue_ignoring_callback_for_older_transaction(
+                    self,
+                    transaction,
+                    value.creationDate
+                ))
+                return nil
```

Comment:
Why does this not need to remove the callback? When would this callback get removed? 🤔 

Just want to make sure its not there for forever 🤷‍♂️

Reply:
Done.